File: 225-implement-stack-using-queues/implement-stack-using-queues.py
from collections import deque

class MyStack(object):

    def __init__(self):
        self.q1 = deque()
        self.q2 = deque()

    def push(self, x):
        self.q2.append(x)
        # Move all elements from q1 to q2
        while self.q1:
            self.q2.append(self.q1.popleft())
        # Swap names of q1 and q2
        self.q1, self.q2 = self.q2, self.q1

    def pop(self):
        return self.q1.popleft()

    def top(self):
        return self.q1[0]

    def empty(self):
        return not self.q1

File: 225-implement-stack-using-queues/implement-stack-using-queues.cpp
class MyStack {
    deque<int> q;
    int size;
public:
    MyStack() : q{}, size{0} {
    }
    
    void push(int x) {
        q.push_front(x);
        size++;
    }
    
    int pop() {
        int i = q[0];
        q.pop_front();
        size--;
        return i;
    }
    
    int top() {
        return q[0];
    }
    
    bool empty() {
        return size == 0;
    }
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */